example-app/src/main/java/com/adyen/checkout/example/ui/googlepay/compose/SessionsGooglePayEvents.kt

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) 2023 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by josephj on 13/12/2023.
+ */
+
+package com.adyen.checkout.example.ui.googlepay.compose
+
+internal abstract class SessionsGooglePayEvents internal constructor() {
```

Comment:
@JlUgia I just noticed this line, is there any benefit from using this over `internal sealed class SessionsGooglePayEvents {`?

Reply:
The main difference is that `sealed class`es enforce exhaustive checking (eg.: a `switch` statement will be forced to include a `case` statement for each member in the class), which has backwards-compatibility implications (especially for lists that are not meant to be exhaustive).

My rule of thumb is to use `sealed class` for exhaustive lists that are always meant to be handled in full. For example view states in a layout (loading, complete, error), and `abstract class` for non-exhaustive lists with unrelated items like `SesssionsGooglePayEvents` (even more so if they are part of an API that is expected to change).

Let me know if that makes sense.

Reply:
This is a really good solution 💯  Especially in library development where adding something to a sealed class can technically cause a breaking change